extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -67,13 +77,109 @@ MinNettyAllocatorMaxOrderBuildItem setMinimalNettyMaxOrderSize() {
+        for (ClassInfo generatedBean : index.getIndex().getAllKnownImplementors(GrpcDotNames.GENERATED_GRPC_BEAN)) {
+            FieldInfo delegateField = generatedBean.field("delegate");
+            if (delegateField == null) {
+                throw new IllegalStateException("A generated bean does not declare the delegate field: " + generatedBean);
+            }
+            DotName generatedInterface = delegateField.type().name();
+            Collection<ClassInfo> serviceCandidates = index.getIndex().getAllKnownImplementors(generatedInterface);
+            if (serviceCandidates.isEmpty()) {
+                // No user-defined bean that implements the generated interface
+                continue;
+            }
+            ClassInfo userDefinedBean = null;
+            for (ClassInfo candidate : serviceCandidates) {
+                // The bean must be annotated with @GrpcService
+                if (candidate.classAnnotation(GrpcDotNames.GRPC_SERVICE) != null) {
+                    userDefinedBean = candidate;
+                    break;
+                }
+            }
+            if (userDefinedBean == null) {
+                continue;
+            }
+            DotName mutinyImplBase = generatedBean.superName();
+            if (index.getIndex().getAllKnownSubclasses(mutinyImplBase).size() != 1) {
+                // Some class extends the mutiny impl base
+                continue;
+            }
+            String mutinyImplBaseName = mutinyImplBase.toString();
+            // Now derive the original impl base
+            // e.g. examples.MutinyGreeterGrpc.GreeterImplBase -> examples.GreeterGrpc.GreeterImplBase
+            DotName implBase = DotName.createSimple(mutinyImplBaseName.replace(MutinyGrpcGenerator.CLASS_PREFIX, ""));
+            if (!index.getIndex().getAllKnownSubclasses(implBase).isEmpty()) {
+                // Some class extends the impl base
+                continue;
+            }
+            // Finally exclude some packages
+            boolean excluded = false;
+            for (String excludedPackage : excludedPackages) {
+                if (mutinyImplBaseName.startsWith(excludedPackage)) {
+                    excluded = true;
+                    break;
+                }
+            }
+            if (!excluded) {
+                logger.debugf("Registering generated gRPC bean %s that will delegate to %s", generatedBean, userDefinedBean);
+                Set<MethodInfo> blockingMethods = new HashSet<>();
+                for (MethodInfo method : userDefinedBean.methods()) {
+                    if (method.hasAnnotation(GrpcDotNames.BLOCKING)) {
+                        blockingMethods.add(method);
+                    }
+                }
+                nameToBlockingMethods.put(generatedBean.name(), blockingMethods);
+            }
+        }
+
+        if (!nameToBlockingMethods.isEmpty()) {
```

Comment:
`nameToBlockingMethods` name made me think that it contains only stuff with blocking methods (which fortunately is not the case :) )